### ISSUE TITLE
Circom 2.1.0 Anonymous Components and Tuples

### ISSUE THREAD

**Issue by vivianjeng** [social-tw/social-tw-website#70]
See: https://docs.circom.io/circom-language/anonymous-components-and-tuples/

Now we assigned every input into another templates with
```circom
template A(n){
   signal input a, b;
   signal output c;
   c <== a*b;
}
template B(n){
   signal input in[n];
   signal out;
   component temp_a = A(n);
   temp_a.a <== in[0]; 
   temp_a.b <== in[1];
   out <== temp_a.c;
}
component main = B(2);
```

with anonymous components, we can assign each input with

```circom
template A(n){
   signal input a, b;
   signal output c;
   c <== a*b;
}
template B(n){
   signal input in[n];
   signal out <== A(n)(in[0],in[1]);
}
component main = B(2);
```

which will improve the readability and reduce lines.

**Comment by FoodChain1028** [comment]
I will deal with this recently.

**Comment by ryanycw** [comment]
Closed with #82
